pex/dist_metadata.py

```diff
@@ -789,14 +789,20 @@ def as_constraint(self):
 class DistMetadata(object):
     @classmethod
     def from_metadata_files(cls, metadata_files):
-        # type: (MetadataFiles) -> DistMetadata
-        return cls(
-            files=metadata_files,
-            project_name=metadata_files.metadata.project_name,
-            version=metadata_files.metadata.version,
-            requires_dists=tuple(requires_dists(metadata_files)),
-            requires_python=requires_python(metadata_files),
-        )
+        project_name = metadata_files.metadata.project_name
+        version = metadata_files.metadata.version
+        try:
+            # type: (MetadataFiles) -> DistMetadata
```

Comment:
Pex supports `>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,<3.14` (See: https://pypi.org/project/pex/ Requires: sidebar); so this serves as a type comment for MyPy that works for Python 2.7, but it only works if its where it was, just under the function `def` line.